CMakeLists.txt

```diff
@@ -219,4 +219,117 @@ if (CONFIG_BUILD_WITH_TFM)
   )
 
   zephyr_link_libraries(tfm_api)
+
+  # Set default image versions if not defined elsewhere
+  if (NOT DEFINED TFM_IMAGE_VERSION_S)
+    set(TFM_IMAGE_VERSION_S 0.0.0+0)
+  endif()
+
+  if (NOT DEFINED TFM_IMAGE_VERSION_NS)
+    set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+  endif()
+
+  if (NOT CONFIG_TFM_BL2_FALSE)
+    set(PREPROCESSED_FILE_S "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o")
+    set(PREPROCESSED_FILE_NS "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o")
+    set(TFM_MCUBOOT_DIR "${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/bl2/ext/mcuboot")
+  endif()
+
+  # Configure which format (full or hash) to include the public key in
+  # the image manifest
+  if(NOT DEFINED TFM_PUBLIC_KEY_FORMAT)
+    set(TFM_PUBLIC_KEY_FORMAT "full")
+  endif()
+
+  if (CONFIG_TFM_BL2_FALSE)
+    # Merge tfm_s and zephyr (NS) image to a single binary.
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/mergehex.py
+        -o ${CMAKE_BINARY_DIR}/tfm_merged.hex
+        $<TARGET_PROPERTY:tfm,TFM_S_HEX_FILE>
+        ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME}
+    )
+
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_byproducts
+      ${CMAKE_BINARY_DIR}/tfm_merged.hex
+    )
+
+  elseif(CONFIG_TFM_MCUBOOT_IMAGE_NUMBER STREQUAL "1")
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/mergehex.py
+        -o ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns.hex
+        $<TARGET_PROPERTY:tfm,TFM_S_HEX_FILE>
+        ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME}
+
+      #Sign concatenated binary image with default public key in mcuboot folder
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/wrapper/wrapper.py
```

Comment:
Have you tried extracting this to a function? It looks like several of the arguments passed to `wrapper.py` are the same across the three invocations.

Reply:
Done